src/operator/tensor/matrix_op.cc

```diff
@@ -233,12 +234,8 @@ static void FlattenEx(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(inputs.size(), 1U);
   CHECK_EQ(outputs.size(), 1U);
 #if MXNET_USE_MKLDNN == 1
-  if (inputs[0].IsMKLDNNData()) {
-    MKLDNNCopy(attrs, ctx, inputs[0], req[0], outputs[0]);
-    // If the output is a special MKLDNN layout and the number of dimensions
-    // is larger than 2, we should use the default layout.
-    if (outputs[0].IsMKLDNNData() && inputs[0].shape().ndim() > 2)
-      const_cast<NDArray &>(outputs[0]).Reorder2Default();
+  if (SupportMKLDNNArray(inputs[0].dtype(), inputs[0].shape())) {
```

Comment:
`SupportMKLDNNArray` doesn't support 3D tensor, `flatten` should have same coverage as `reshape`, right?

Reply:
Yes, you're right.

Reply:
Use the same conditions in `SupportMKLDNNReshape`.